src/components/GitPanel.tsx

```diff
@@ -205,67 +217,36 @@ export class GitPanel extends React.Component<IGitPanelProps, IGitPanelState> {
   };
 
   /**
-   * Commits all marked files.
+   * Commits files.
    *
-   * @param message - commit message
-   * @returns a promise which commits the files
+   * @returns a promise which commits changes
    */
-  commitMarkedFiles = async (message: string): Promise<void> => {
-    this.props.logger.log({
-      level: Level.RUNNING,
-      message: this.props.trans.__('Staging files...')
-    });
-    await this.props.model.reset();
-    await this.props.model.add(...this._markedFiles.map(file => file.to));
+  commitFiles = async (): Promise<void> => {
+    let msg = this.state.commitSummary;
 
-    await this.commitStagedFiles(message);
-  };
+    // Only include description if not empty
+    if (this.state.commitDescription) {
+      msg = msg + '\n\n' + this.state.commitDescription + '\n';
+    }
 
-  /**
-   * Commits all staged files.
-   *
-   * @param message - commit message
-   * @returns a promise which commits the files
-   */
-  commitStagedFiles = async (message: string): Promise<void> => {
-    if (!message) {
+    if (!msg) {
       return;
     }
```

Comment:
As you know, the CommitBox was responsible for the commit message, and one thing I came across was 

This check initially, never actually did anything, since `message` before my changes, will never be empty during this check.
What I mean by that is,
```ts
const summary = "";
const description = ""
const msg = summary + '\n\n' + description + '\n';
console.log(!msg);  // false   
```

This wasn't picked up from a user standpoint since the commit button disables/enables correctly. But yea just fyi I guess

Reply:
Thanks for the detailed explanation and good catch.